Prefetching-Library/android_prefetching_lib/src/main/java/nl/vu/cs/s2group/nappa/prefetch/PrefetchStrategy.java

```diff
@@ -2,12 +2,75 @@
 
 import androidx.annotation.NonNull;
 
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+
 import java.util.List;
 
 import nl.vu.cs.s2group.nappa.graph.ActivityNode;
 
 public interface PrefetchStrategy {
+    @Deprecated
+    int STRATEGY_1 = 1;
+    @Deprecated
+    int STRATEGY_2 = 2;
+    int STRATEGY_GREEDY = 3;
+    @Deprecated
+    int STRATEGY_4 = 4;
+    @Deprecated
+    int STRATEGY_5 = 5;
+    @Deprecated
+    int STRATEGY_6 = 6;
+    @Deprecated
+    int STRATEGY_7 = 7;
+    @Deprecated
+    int STRATEGY_8 = 8;
+    @Deprecated
+    int STRATEGY_9 = 9;
 
-    @NonNull List<String> getTopNUrlToPrefetchForNode(ActivityNode node, Integer maxNumber);
+    /**
+     * Obtain the most likely {@link android.app.Activity} the user is likely to navigate to
+     *
+     * @param node      Represents the current {@link android.app.Activity} to where the user navigated to
+     * @param maxNumber Limits the number of requests to prefetch
+     * @return A list of URLs with high potential to be requested by the user in the immediate future
+     */
+    @NonNull
+    List<String> getTopNUrlToPrefetchForNode(ActivityNode node, Integer maxNumber);
 
+    /**
+     * Instantiate the prefetching strategy corresponding to the provided ID.
+     * If the ID is unknown, instantiate the default Greedy-based strategy implemented by
+     * {@link GreedyPrefetchStrategy}
+     *
+     * @param strategyId The identification number of the prefetching strategy.
+     * @return A implemented prefetching strategy.
+     */
+    @SuppressWarnings("DuplicateBranchesInSwitch")
+    @NotNull
+    @Contract("_ -> new")
+    static PrefetchStrategy getStrategy(int strategyId) {
```

Comment:
Having a number for identifying a strategy forces us to always remember the mapping between the number and the strategy. What about having an enumeration, so that the strategy identifiers can be mnemonic names?

Reply:
Yes, having to remember the mapping ID ~ Strategy is not ideal.
For starter, I have defined constants in this interface, one for each strategy implementation.
Then instead of informing the actual ID, developers need to inform the constant when initializing the library in the target app:

```java
PrefetchingLib.init(PrefetchStrategy.STRATEGY_GREEDY)
```

After exploring the source-code and debugging the application running this last week, I have the following:

* Strategy 5 and 8 uses PageRank
* Strategy 6 and 9 uses an algorithm named HITS
* Strategy 7 uses an algorithm named SALSA

I am not sure what is the difference between 5 ~ 8 and 6 ~ 9, but scores for PageRank and HITS are calculated in both strategies. Similarly, I don't know about strategies 1, 2 and 4.

Reply:
I can already update the name for SALSA and, while not ideal, I can rename `S5` and `S8` to `PageRank` and `PageRank2`. Similar to `S9` and `S9` to `HITS` and `HITS2`
Once it is identified what is the difference between these strategies, we can use better names.

Reply:
Okay, I did some further digging and I think that I got the differences of the PageRank and HITS strategies and strategies 1, 2 and 4:

* Strategy 5 seems to use plain PageRank scores
* Strategy 8 seems to be a mix of the Greedy and strategy 5, where the overall logic uses the Greedy strategy but uses the PageRank score as the probability score.
* Strategy 4 seems to use partial match and a Markov model
* Strategy 6 seems to use plain HITS scores
* Strategy 9 seems to be a mix of strategies 4 and 6, where the overall logic uses the partial match from strategy 4 but uses the HITS score as the probability score.
* Strategy 1 seems to select the successor with most visits
* I am not sure how Strategy 2 works, but it seems to use a text manipulation library to do some operations with the URL. I could not find anything that resembled a score calculation to select nodes.

With this, we can use some proper namings for most strategies

Reply:
Ok, give a proper mnemonic name to each strategy and let's proceed!